### DISCUSSION TITLE
how to load multiple geotiffs with deiiferent resolutions  and read label(for training) stored in one band for each tile?

### DISCUSSION THREAD

**Discussion by mohscorpion** [torchgeo/torchgeo#1700] [Q&A]
hi
i try to use torchgeo in my project, i have multiple geotiff with different resolution and different bands.
how can load all in one dataset and use value of a band as training label?
i notice samplers do tile up the loaded dataset but i can't find hot to load the label for training model.
thanks

**Comment by adamjstewart** [comment]
Just want to make sure you've seen https://torchgeo.readthedocs.io/en/stable/tutorials/custom_raster_dataset.html, which documents how to write a custom RasterDataset. Obviously your task is a bit harder, as you're trying to load both images and labels.

One was is to make two RasterDatasets, one for images and one for labels. Then you can compute the intersection. The other way is to make one RasterDataset that does both. The L7Irish/L8Biome/LandCoverAIGeo datasets are good examples of this.

If your data is already pre-chipped and you don't need the ability to combine it with other datasets, it may be simpler to write a NonGeoDataset instead. There are 50+ examples of this built in to TorchGeo.

Hope that helps!
